### DISCUSSION TITLE
What is the difference between IsSecure and IsEncrypted?

### DISCUSSION THREAD

**Discussion by rklec** [jstedfast/MailKit#1717] [Q&A]
I am using the `ImapClient` but this question may be answered more generally as it is applicable to all clients as far as I see.
There is IsSecure and IsEncrypted, but both have nearly the same doc/description, so what is actually the difference between them?
When should I use what (e.g. to check for an encrypted connection as in https://github.com/jstedfast/MailKit/issues/387).

IsSecure:
https://github.com/jstedfast/MailKit/blob/ff722f1a0967cb2118d109a5a2136dff2c793ac4/MailKit/Net/Imap/ImapClient.cs#L684-L693

IsEncrypted:
https://github.com/jstedfast/MailKit/blob/ff722f1a0967cb2118d109a5a2136dff2c793ac4/MailKit/Net/Imap/ImapClient.cs#L695-L704

Only the code itself for `IsEncrypted` hints, it does more.  **So a connection is encrypted when it is (also) secure?** Note for my language understanding thus sounds weird, as usually, I'd say a connection is secure, when it is encrypted (and maybe authenticated etc.). So "security" as a concept involves more than just encryption. But that's just my personal opinion here.
**However**, this only seems to apply to the `ImapClient`. So what is the difference, in human terms?

**Comment by jstedfast** [comment]
The IsSecure property was added first and all it means is that an SSL/TLS connection has been made.

IsEncrypted was added years later and returns true if the SslStream's IsEncrypted property is true.
